File: TwoPointer/66.plus-one.py
from typing import List


# @leet start
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        for i in range(i, -1, -1):
            if digits[i] + 1 != 10:
                digits[i] += 1
                return digits
            digits[i] = 0
            # digits[i - 1] += 1 # NOTE: double dipple
        if i == 0:
            return [1] + digits
        else:
            return digits


# @leet end

File: TwoPointer/66.plus-one.go
package leetcode

// @leet start
func plusOne(digits []int) []int {
	n := len(digits) - 1
	for ; n >= 0; n-- {
		if digits[n]+1 != 10 {
			digits[n]++
			return digits
		}
		digits[n] = 0
		if n == 0 {
			// ret := []int{1}
			// ret = append(ret, digits...)
			return append([]int{1}, digits...)
			// return ret
		}
	}
	// if n == -1 {
	// 	// ret := []int{1}
	// 	// ret = append(ret, digits...)
	// 	return append([]int{1}, digits...)
	// 	// return ret
	// }
	return digits
}

// @leet end

